indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx

```diff
@@ -73,32 +70,39 @@ class BookingEdit extends React.Component {
     constructor(props) {
         super(props);
 
-        const {booking: {occurrences, dateRange}} = props;
         this.state = {
             skipConflicts: false,
             numberOfConflicts: null,
             numberOfCandidates: null,
-            shouldSplit: false,
+            willBookingSplit: false,
+            calendars: null,
+        };
+    }
+
+    async componentDidMount() {
+        // we need to use setState here to rerender the
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({calendars: this.initialCalendarData});
+
+        const {dates, timeSlot, recurrence} = this.formInitialValues;
+        const {
+            calendars: [currentBooking]
+        } = await this.fetchBookingTimelineInfo({dates, timeSlot, recurrence});
+
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({
+            isLoading: false,
             calendars: {
-                currentBooking: {
-                    isFetching: false,
-                    data: {
-                        bookings: {...occurrences.bookings},
-                        cancellations: {...occurrences.cancellations},
-                        rejections: {...occurrences.rejections},
-                        other: {...occurrences.otherBookings},
-                        blockings: {...occurrences.blockings},
-                        overridableBlockings: {...occurrences.overridableBlockings},
-                        nonbookablePeriods: {...occurrences.nonbookablePeriods},
-                        unbookableHours: occurrences.unbookableHours,
-                        pendingCancellations: {},
-                        candidates: {},
-                        conflictingCandidates: {},
-                        conflicts: {}
-                    },
-                    dateRange,
-                },
-                newBooking: null, // available only when splitting an ongoing booking
+                currentBooking,
+            }
+        });
+    }
+
+    get initialCalendarData() {
```

Comment:
assuming you never mutate this object, why not just make it a `const` outside the component, or a static attribute?